### PR TITLE
Using IMalloc::Free instead of delete for Release.

### PR DESCRIPTION
**Pull request by adam-yang**

### CODE REVIEW

File: include/dxc/Support/microcom.h [hunk 1]
```diff
@@ -104,7 +109,8 @@ inline T *CreateOnMalloc(IMalloc * pMalloc, Args&&... args) {
       ULONG result = InterlockedDecrement(&m_dwRef); \
       if (result == 0) { \
         CComPtr<IMalloc> pTmp(m_pMalloc); \
-        DxcThreadMalloc M(pTmp); delete this; \
```
Comment (marcelolr) [inline include/dxc/Support/microcom.h]: I think you still need to set up the DxcThreadMalloc, for the cases where sub-objects within this need to have it set up. It's implied that an object with its own IMalloc will use it for internal objects, even if those are new/delete-based.

Comment (adam-yang) [reply include/dxc/Support/microcom.h]: I added it back in. Also fixed a class that does not go through the usual microcom construction that sets the IMalloc member. All tests are passing now.

File: output.txt [hunk 2]
```diff
@@ -0,0 +1,3612 @@
+Copying binaries to test to C:\Users\hanbyang\Documents\sandbox\hlsl.bin\Debug\test:
```
Comment (youngkim93) [inline output.txt]: Is this file added intentionally?

File: output2.txt [hunk 3]
```diff
@@ -0,0 +1,3608 @@
+Copying binaries to test to C:\Users\hanbyang\Documents\sandbox\hlsl.bin\Debug\test:
```
Comment (youngkim93) [inline output2.txt]: Same here

**Pull request merged**